tests/unit/lib/SGSparseMatrix_unittest.cc

```diff
@@ -27,18 +27,16 @@ using namespace Eigen;
 
 //function can generate both sparse and dense matrix accroding to sparse convention
 template<class MatrixType>
-MatrixType GenerateMatrix(float64_t sparseLevel, int32_t m, int32_t n, int32_t randSeed)
+void GenerateMatrix(float64_t sparseLevel, int32_t m, int32_t n, int32_t randSeed, MatrixType* matrix)
```

Comment:
SG dense matrix has no initialization upon construction, thus the template function cannot generate both sparse and dense matrix with zeros initialized. Thus I choose to initialize the matrix outside the function, and pass in it as an argument to set the nonzero entries.